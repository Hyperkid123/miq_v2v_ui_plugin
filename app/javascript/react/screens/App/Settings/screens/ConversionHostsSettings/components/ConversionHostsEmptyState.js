import React from 'react';
import PropTypes from 'prop-types';
import ShowWizardEmptyState from '../../../../common/ShowWizardEmptyState/ShowWizardEmptyState';

const ConversionHostsEmptyState = ({ showConversionHostWizardAction }) => (
  <ShowWizardEmptyState
    description={
      __('Select one or more hosts to be configured as IMS conversion hosts.') // prettier-ignore
    }
    showWizardAction={showConversionHostWizardAction}
    buttonText={__('Configure Conversion Host')}
    buttonHref=""
    className="white-bg"
  />
);

ConversionHostsEmptyState.propTypes = {
  showConversionHostWizardAction: PropTypes.func
};

export default ConversionHostsEmptyState;
