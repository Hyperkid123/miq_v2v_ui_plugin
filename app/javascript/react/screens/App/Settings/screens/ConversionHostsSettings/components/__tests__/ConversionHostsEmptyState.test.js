import React from 'react';
import { shallow, mount } from 'enzyme';
import ConversionHostsEmptyState from '../ConversionHostsEmptyState';

describe('conversion hosts empty state', () => {
  const getBaseProps = () => ({
    showConversionHostWizardAction: jest.fn()
  });

  it('renders correctly', () => {
    const component = shallow(<ConversionHostsEmptyState {...getBaseProps()} />);
    expect(component).toMatchSnapshot();
  });

  it('calls the correct action on button click', () => {
    const props = getBaseProps();
    const component = mount(<ConversionHostsEmptyState {...props} />);
    component.find('Button').simulate('click');
    expect(props.showConversionHostWizardAction).toHaveBeenCalledTimes(1);
  });
});
