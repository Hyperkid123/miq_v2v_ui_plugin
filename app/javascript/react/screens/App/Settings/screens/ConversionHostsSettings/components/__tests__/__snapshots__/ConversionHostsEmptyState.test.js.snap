// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`conversion hosts empty state renders correctly 1`] = `
<ShowWizardEmptyState
  buttonHref=""
  buttonText="Configure Conversion Host"
  className="white-bg"
  description="Select one or more hosts to be configured as IMS conversion hosts."
  iconName="add-circle-o"
  iconType="pf"
  showWizardAction={[MockFunction]}
  title=" "
/>
`;
